harper-core/src/linting/that_than.rs

```diff
@@ -0,0 +1,240 @@
+                !is_better_or_later
+            })
+            .t_ws()
+            .t_aco("that")
+            .t_ws()
+            // .then_any_word();
+            .then(|tok: &Token, src: &[char]| {
+                if !tok.kind.is_word() {
+                    return false;
+                }
+                let adj_lower = tok.span.get_content_string(src).to_lowercase();
+                let is_way = adj_lower == "way";
+                !is_way
+            });
+
+        Self {
+            expr: Box::new(adjective_er_that_nextword),
+        }
+    }
+}
+
+impl ExprLinter for ThatThan {
+    fn expr(&self) -> &dyn Expr {
+        self.expr.as_ref()
+    }
+
+    fn match_to_lint(&self, toks: &[Token], src: &[char]) -> Option<Lint> {
+        if toks.len() != 5 {
+            return None;
+        }
+
+        let that_tok = &toks[2];
+
+        Some(Lint {
+            span: that_tok.span,
+            lint_kind: LintKind::Typo,
+            suggestions: vec![Suggestion::replace_with_match_case_str(
+                "than",
+                that_tok.span.get_content(src),
+            )],
+            message: "This looks like a comparison that should use `than` rather than `that`."
+                .to_string(),
+            priority: 31,
+        })
+    }
+
+    fn description(&self) -> &'static str {
+        "Corrects the typo `that` to `than` in comparisons."
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::linting::{
+        ThatThan,
+        tests::{assert_lint_count, assert_suggestion_result},
+    };
+
+    // adj-er that
+
+    #[test]
+    fn fix_slower_that() {
+        assert_suggestion_result(
+            "Local installed PHAR 5x times slower that the same PHAR installed globally",
+            ThatThan::default(),
+            "Local installed PHAR 5x times slower than the same PHAR installed globally",
+        );
+    }
+
+    #[test]
+    fn dont_flag_more_that() {
+        assert_lint_count(
+            "so it's probably more that Croatian had an easier test",
+            ThatThan::default(),
+            0,
+        );
+    }
+
+    #[test]
+    fn dont_flag_easier_that_way() {
+        assert_lint_count(
+            "Given svelte now has signals, it might actually be easier that way.",
+            ThatThan::default(),
+            0,
+        );
+    }
+
+    #[test]
+    fn dont_flag_better_that() {
+        assert_lint_count(
+            "So I am wondering if its better that I run SCENIC+ once on the integrated dataset or 3 times on the individual datasets",
+            ThatThan::default(),
+            0,
+        );
+    }
+
+    #[test]
+    #[ignore = "not handled because 'better' results in false positives"]
+    fn fix_better_that() {
+        assert_suggestion_result(
+            "Examples of how different cards perform far better that others.",
+            ThatThan::default(),
+            "Examples of how different cards perform far better than others.",
+        );
+    }
```

Comment:
See https://github.com/Automattic/harper/pull/1639#discussion_r2233747831

I feel like detecting this in ThenThan could be interesting

            "Examples of how different cards perform far better then others."